variables.tf

```diff
@@ -129,3 +129,63 @@ variable "content_based_deduplication" {
   type        = bool
   default     = false
 }
+
+variable "create_sns_topic_subscription" {
+  description = "Whether to create the SNS topic subscription"
+  type        = bool
+  default     = false
+}
+
+variable "endpoint" {
+  description = "Endpoint to send data to"
+  type        = string
+  default     = null
+}
+
+variable "protocol" {
+  description = "Protocol to use"
+  type        = string
+  default     = null
+}
+
+variable "topic_arn" {
+  description = "ARN of the SNS topic to subscribe to"
+  type        = string
+  default     = null
+}
+
+variable "confirmation_timeout_in_minutes" {
+  description = "Integer indicating number of minutes to wait in retrying mode for fetching subscription arn before marking it as failure"
+  type        = number
+  default     = 1
+}
+
+# variable "delivery_policy" {
```

Comment:
There is a naming conflict and I am not sure if SNS topic and SNS topic subscriptions use different delivery policies.